server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -78,6 +82,10 @@ public class ShuffleServer {
   private AtomicBoolean isHealthy = new AtomicBoolean(true);
   private GRPCMetrics grpcMetrics;
   private MetricReporter metricReporter;
+  private Thread decommissionThread;
+  private ServerStatus serverStatus = ServerStatus.NORMAL_STATUS;
```

Comment:
This status should be marked as `volatile,` which will be used in different threads. Right

Reply:
Right